### PR TITLE
Fix: don't crash on empty alert rules file

### LINKED ISSUE #364: Empty alert rules files errors out
### Bug Description

The `AlertRules._from_file()` method errors out when adding an empty (zero-size) alert rule file, see below for traceback

### To Reproduce

1. Deploy a client of the `prometheus_scrape` library
2. Create an empty alert.rules file: `touch alert.rules`
3. Trigger reloading alert rule parsing

### Environment

Platform-indep.

### Relevant log output

```shell
Traceback (most recent call last):
  File "/home/peter/src/cos-lite/prometheus-k8s-operator/tests/unit/test_prometheus_rules_provider.py", line 161, in test_reload_with_empty_rules
    self.harness.charm.rules_provider._reinitialize_alert_rules()
  File "/home/peter/src/cos-lite/prometheus-k8s-operator/lib/charms/prometheus_k8s/v0/prometheus_scrape.py", line 1684, in _reinitialize_alert_rules
    self._update_relation_data(None)
  File "/home/peter/src/cos-lite/prometheus-k8s-operator/lib/charms/prometheus_k8s/v0/prometheus_scrape.py", line 1692, in _update_relation_data
    alert_rules.add_path(self.dir_path, recursive=self._recursive)
  File "/home/peter/src/cos-lite/prometheus-k8s-operator/lib/charms/prometheus_k8s/v0/prometheus_scrape.py", line 762, in add_path
    self.alert_groups.extend(self._from_dir(path, recursive))
  File "/home/peter/src/cos-lite/prometheus-k8s-operator/lib/charms/prometheus_k8s/v0/prometheus_scrape.py", line 740, in _from_dir
    alert_groups_from_file = self._from_file(dir_path, file_path)
  File "/home/peter/src/cos-lite/prometheus-k8s-operator/lib/charms/prometheus_k8s/v0/prometheus_scrape.py", line 644, in _from_file
    if _is_official_alert_rule_format(rule_file):
  File "/home/peter/src/cos-lite/prometheus-k8s-operator/lib/charms/prometheus_k8s/v0/prometheus_scrape.py", line 563, in _is_official_alert_rule_format
    return "groups" in rules_dict
TypeError: argument of type 'NoneType' is not iterable
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by sabaini**
## Issue
Fixes #364

## Solution
Add defensive check in AlertRules._from_file()

### CODE REVIEW

File: lib/charms/prometheus_k8s/v0/prometheus_scrape.py [hunk 1]
```diff
@@ -641,6 +641,10 @@ def _from_file(self, root_path: Path, file_path: Path) -> List[dict]:
                 logger.error("Failed to read alert rules from %s: %s", file_path.name, e)
                 return []
 
+            if not rule_file or not isinstance(rule_file, dict):
+                logger.error("Rule file empty or not a dictionary: %s", file_path.name)
+                return []
+
```
Comment (sed-i) [inline lib/charms/prometheus_k8s/v0/prometheus_scrape.py:647]:
```suggestion
            if not rule_file:
                logger.warning("Empty rule file: %s", file_path.name)
                return []
            if not isinstance(rule_file, dict):
                logger.error("Invalid rules file (must be a dict): %s", file_path.name)
                return []
```

**Review by sed-i — COMMENTED** [review]
Thanks! Looks good.
A minor suggestion.

**Comment by sabaini** [comment]
Thanks, suggestion lgtm!

**Review by rbarry82 — APPROVED** [review]

File: lib/charms/prometheus_k8s/v0/prometheus_scrape.py [hunk 2]
```diff
@@ -641,6 +641,12 @@ def _from_file(self, root_path: Path, file_path: Path) -> List[dict]:
                 logger.error("Failed to read alert rules from %s: %s", file_path.name, e)
                 return []
 
+            if not rule_file:
+                logger.warning("Empty rule file: %s", file_path.name)
```
Comment (beliaev-maksim) [inline lib/charms/prometheus_k8s/v0/prometheus_scrape.py:645]: just to be consistent with the error message in another leaf
```suggestion
                logger.warning("Empty rules file: %s", file_path.name)
```

PS
I would even prefer initial implementation with a single check :)
but do not have a strong opinion on this

**Review by sed-i — APPROVED** [review]

**Pull request merged**